### PR TITLE
[harvest] Do not delete existing item images

### PR DESCRIPTION
**Pull request by rastislav-chynoransky**
Fixes https://jira.sng.sk/browse/WEBUMENIA-1405

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] I have updated the [CHANGELOG](../CHANGELOG.md)
- [x] I have requested at least 1 reviewer for this PR
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: app/Harvest/Importers/AbstractImporter.php [hunk 1]
```diff
@@ -109,19 +110,21 @@ protected function processHasMany(Model $model, $field, array $relatedRows) {
...
 
-        $relatedKeyName = $relation->getRelated()->getKeyName();
-        $relation->whereNotIn($relatedKeyName, $updateIds)->each(function (Model $related) {
-            $related->delete();
-        });
+        if ($allowDelete) {
+            $relatedKeyName = $relation->getRelated()->getKeyName();
+            $relation->whereNotIn($relatedKeyName, $updateIds)->each(function (Model $related) {
+                $related->delete();
+            });
```
Comment (eronisko) [inline app/Harvest/Importers/AbstractImporter.php:117]: Would this also work?
```suggestion
            $relation->whereNotIn($relatedKeyName, $updateIds)->delete();
```

File: tests/Harvest/Harvesters/ItemHarvesterTest.php [hunk 2]
```diff
@@ -14,6 +14,7 @@
 use App\Harvest\Repositories\ItemRepository;
 use App\Harvest\Result;
 use App\Item;
+use App\ItemImage;
```
Comment (eronisko) [inline tests/Harvest/Harvesters/ItemHarvesterTest.php:17]: Does not seem to be used here
```suggestion
```

**Review by eronisko — APPROVED** [review]
Looks good.!

Out of curiosity: could the `processBelongsToMany` and `processHasMany` methods be covered by Laravel's Eloquent relationships sometime in the future? (https://laravel.com/docs/7.x/eloquent-relationships)

**Comment by rastislav-chynoransky** [comment]
> Looks good.!
> 
> Out of curiosity: could the `processBelongsToMany` and `processHasMany` methods be covered by Laravel's Eloquent relationships sometime in the future? (https://laravel.com/docs/7.x/eloquent-relationships)

I'd love to reduce unnecessary code on our side, but I'm not sure what do you mean by that.

File: app/Harvest/Importers/AbstractImporter.php (hunk 1 above)
Comment (rastislav-chynoransky) [reply app/Harvest/Importers/AbstractImporter.php:117]: The case of bulk delete model events wouldn't be fired
https://laravel.com/docs/7.x/eloquent#deleting-models

**Pull request merged**